### PR TITLE
Update applications API use to pass access token as body param instead of path param

### PR DESCRIPTION
**Pull request by oreoshake**
See https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api and https://developer.github.com/changes/2020-02-14-deprecating-oauth-app-endpoint/

This PR updates the calls to the applications API calls. The current implementation uses deprecated endpoint, the new implementation uses the supported, but preview endpoints.

I've got the tests passing but I've yet to actually try this against the live API :smile:

### CODE REVIEW

File: lib/octokit/client/authorizations.rb [hunk 1]
```diff
@@ -152,12 +152,13 @@ def scopes(token = @access_token, options = {})
...
       def check_application_authorization(token, options = {})
-        opts = options.dup
+        opts = ensure_api_media_type(:applications_api, options.dup)
+        opts[:access_token] = token
         key    = opts.delete(:client_id)     || client_id
         secret = opts.delete(:client_secret) || client_secret
 
         as_app(key, secret) do |app_client|
-          app_client.get "applications/#{client_id}/tokens/#{token}", opts
+          app_client.post "applications/#{client_id}/tokens", opts
```
Comment (augustocravosilva) [inline lib/octokit/client/authorizations.rb]: Hey @tarebyte,
The paths you are using here and returing 404. According to https://developer.github.com/changes/2020-02-14-deprecating-oauth-app-endpoint/#endpoints-affected , it should be singular `token`, and not `tokens`

Comment (tarebyte) [reply lib/octokit/client/authorizations.rb]: Damn, thanks for the heads up. I'm working on a PR to resolve this.

Comment (oreoshake) [reply lib/octokit/client/authorizations.rb]: > I've got the tests passing but I've yet to actually try this against the live API 😄

🤦‍♂️

Comment (tarebyte) [reply lib/octokit/client/authorizations.rb]: https://github.com/octokit/octokit.rb/releases/tag/v4.18.0

Comment (augustocravosilva) [reply lib/octokit/client/authorizations.rb]: Thanks for the quick fix. Seems to be working correctly now :)

**Comment by emma-sax4** [comment]
@tarebyte Does this change mean that the newest version of Octokit will address this: https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/?

**Comment by tarebyte** [comment]
> Does this change mean that the newest version of Octokit will address this: https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/?

Yup that should be it!

**Pull request merged**